test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -894,39 +894,109 @@ public async Task VerifyRestorePackageByRestoreNuGetPackagesContextMenu(ProjectT
             CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
 
             // Assert
-            CommonUtility.WaitForDirectoryExists(installedPackageFolderPath);
+            CommonUtility.AssertDirectoryExists(installedPackageFolderPath, true);
         }
 
-        [DataTestMethod]
+        [TestMethod]
+        [DataRow(ProjectTemplate.NetCoreConsoleApp)]
+        [DataRow(ProjectTemplate.ConsoleApplication)]
         [Timeout(DefaultTimeout)]
-        public async Task VerifyRestorePackageByBuilding()
+        public async Task VerifyPackageRestoredByBuilding(ProjectTemplate projectTemplate)
         {
             // Arrange
+            string packageFolderPath;
+            string installedPackageFolderPath;
             await CommonUtility.CreatePackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1);
 
             NuGetApexTestService nugetTestService = GetNuGetTestService();
 
             SolutionService solutionService = VisualStudio.Get<SolutionService>();
             solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
-            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V48, "TestProject");
+            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, "TestProject");
             VisualStudio.ClearOutputWindow();
             solutionService.SaveAll();
 
             CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
             NuGetUIProjectTestExtension uiwindow = nugetTestService.GetUIWindowfromProject(project);
             uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
 
-            var packageFolderPath = _pathContext.PackagesV2;
-            var installedPackageFolderPath = Path.Combine(packageFolderPath, "Contoso.A.1.0.0");
+            if (projectTemplate.Equals(ProjectTemplate.NetCoreConsoleApp))
+            {
+                packageFolderPath = _pathContext.UserPackagesFolder;
+                installedPackageFolderPath = Path.Combine(packageFolderPath, TestPackageName);
+            }
+            else
+            {
+                packageFolderPath = _pathContext.PackagesV2;
+                installedPackageFolderPath = Path.Combine(packageFolderPath, "Contoso.A.1.0.0");
+            }
+
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
 
-            // Act
+            // Act (we doesn't edit the "nuget.config" file by setting the "automatic" to "true" since it is enabled by default)
```

Comment:
We added this comment because we wanted to explain why we didn't edit the "nuget.config" file, if you think it's unnecessary, we will delete this comment.

Reply:
The test name does not really suggest that any nuget.config editing is necessary.